guides/source/association_basics.md

```diff
@@ -355,2288 +447,2474 @@ class CreateAccountHistories < ActiveRecord::Migration[7.2]
+This is because creating an association with a name that clashes with an
+existing method could lead to unintended consequences, such as overriding the
+base method and causing issues with functionality. For example, using names like
+`attributes` or `connection` for associations would be problematic.
 
-```ruby
-class Author < ApplicationRecord
-  has_many :books
-end
+### Updating the Schema
 
-class Book < ApplicationRecord
-  belongs_to :author
-  has_many :chapters
-end
+Associations are extremely useful, they are responsible for defining the
+relationships between models but they do not update your database schema. You
+are responsible for maintaining your database schema to match your associations.
+This usually involves two main tasks: creating foreign keys for [`belongs_to`
+associations](#belongs-to) and setting up the correct join table for [`has_many
+:through`](#has-many-through) and
+[`has_and_belongs_to_many`](#has-and-belongs-to-many) associations. You can read
+more about when to use a `has_many :through vs has_and_belongs_to_many` [in the
+has many through vs has and belongs to many
+section](#has-many-through-vs-has-and-belongs-to-many).
 
-class Chapter < ApplicationRecord
-  belongs_to :book
-end
-```
+#### Creating Foreign Keys for `belongs_to` Associations
 
-If you frequently retrieve chapters directly from authors (`@author.books.chapters`), then you can make your code somewhat more efficient by including chapters in the association from authors to books:
+When you declare a [`belongs_to` association](#belongs-to), you need to create
+foreign keys as appropriate. For example, consider this model:
 
 ```ruby
-class Author < ApplicationRecord
-  has_many :books, -> { includes :chapters }
-end
-
 class Book < ApplicationRecord
   belongs_to :author
-  has_many :chapters
 end
+```
+
+This declaration needs to be backed up by a corresponding foreign key column in
+the books table. For a brand new table, the migration might look something like
+this:
 
-class Chapter < ApplicationRecord
-  belongs_to :book
+```ruby
+class CreateBooks < ActiveRecord::Migration[7.2]
+  def change
+    create_table :books do |t|
+      t.datetime   :published_at
+      t.string     :book_number
+      t.belongs_to :author
+    end
+  end
 end
 ```
 
-##### `limit`
-
-The `limit` method lets you restrict the total number of objects that will be fetched through an association.
+Whereas for an existing table, it might look like this:
 
 ```ruby
-class Author < ApplicationRecord
-  has_many :recent_books,
-    -> { order('published_at desc').limit(100) },
-    class_name: "Book"
+class AddAuthorToBooks < ActiveRecord::Migration[7.2]
+  def change
+    add_reference :books, :author
+  end
 end
 ```
 
-##### `offset`
 
-The `offset` method lets you specify the starting offset for fetching objects via an association. For example, `-> { offset(11) }` will skip the first 11 records.
+#### Creating Join Tables for `has_and_belongs_to_many` Associations
```

Comment:
I like that we have this section in the association guides because it's very specific to the relationship and makes it clear that you need the right schema. It’s also easy to see what needs to be done, and everything is in one place. I added a link to the [Active Record Migrations Guides](https://edgeguides.rubyonrails.org/active_record_migrations.html#creating-a-join-table) so people can dive deeper into creating a join table. But I’d like to keep this section here too, unless you feel really strongly about removing it, then I can?

Reply:
That makes sense. If we are keeping this section, I think the following warning about model precedence in table naming should probably be moved to the migration guide.
I think it's too detailed for this section and it isn't mentioned in the migrations guide.

The warning is a bit unclear as well with the lexical scope ordering, while we mention alphabetical ordering elsewhere.

Reply:
Totally agree. I had [moved the warning to the migrations docs yesterday and changed the term in active record migrations to lexical which seems more accurate based on the warning](https://github.com/rails/rails/pull/52523/commits/4d45a7e593f2a88167b2efe99973063c4575eca1) but I'd missed removing the warning from here, which was my intention. Removing it now.

Reply:
Oh wait, no I did actually remove it already ( I was looking at my active record validations branch), here's the [commit](https://github.com/rails/rails/pull/52523/commits/4d45a7e593f2a88167b2efe99973063c4575eca1)

Reply:
I've created a PR the add more sorting examples the the API docs:
https://github.com/rails/rails/pull/52766

Reply:
Ah nevermind. You already made the changes 🙃

Reply:
This is looking great! I'll have a last quick scan through the whole guide later today and I'll merge it.

Reply:
Thank you so much!